module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -772,243 +801,113 @@ namespace app
         sender->bus.sendUnicast(msg, application);
     }
 
-    void ApplicationCommon::handlePhoneModeChanged(sys::phone_modes::PhoneMode mode)
+    void ApplicationCommon::actionPopupPush(std::unique_ptr<gui::SwitchData> params)
     {
-        auto flightModeSetting = settings->getValue(settings::Cellular::offlineMode, settings::SettingsScope::Global);
-        bool flightMode        = flightModeSetting == "1" ? true : false;
-
-        using namespace gui::popup;
-        const auto &popupName = resolveWindowName(gui::popup::ID::PhoneModes);
-        if (const auto currentWindowName = getCurrentWindow()->getName(); currentWindowName == popupName) {
-            updateCurrentWindow(std::make_unique<gui::ModesPopupData>(mode, flightMode));
+        auto rdata = dynamic_cast<gui::PopupRequestParams *>(params.get());
+        if (rdata == nullptr) {
+            assert(0 && "this should never happen");
```

Comment:
hmm change that to some more meaningful :)

Reply:
What should I add here?

Reply:
ok, what do you want to have here then?

Reply:
change to: "invalid popup data recieved"